lib/pbench/test/functional/agent/cli/commands/tools/test_tools_list.py

```diff
@@ -1,31 +1,106 @@
 import pytest
 
 
-def test_pbench_list_tools_help():
-    command = ["pbench-list-tools", "--help"]
-    out, err, exitcode = pytest.helpers.capture(command)
-    assert b"Usage: pbench-list-tools [OPTIONS]" in out
-    assert exitcode == 0
-
-
-def test_list_tool(monkeypatch, agent_config, pbench_run, pbench_cfg):
-    monkeypatch.setenv("_PBENCH_AGENT_CONFIG", str(pbench_cfg))
-    p = pbench_run / "tools-v1-default" / "testhost.example.com"
-    p.mkdir(parents=True)
-    tool = p / "perf"
-    tool.touch()
-
-    command = ["pbench-list-tools"]
-    out, err, exitcode = pytest.helpers.capture(command)
-    assert exitcode == 0
-    assert b"default: testhost.example.com ['perf']" in out
-
-    command = ["pbench-list-tools", "--group", "default"]
-    out, err, exitcode = pytest.helpers.capture(command)
-    assert exitcode == 0
-    assert b"default: testhost.example.com ['perf']" in out
-
-    command = ["pbench-list-tools", "-n", "perf"]
-    out, err, exitcode = pytest.helpers.capture(command)
-    assert exitcode == 0
-    assert b"tool name: perf groups: default" in out
+empty = b""
+usage = b"Usage: pbench-list-tools [OPTIONS]"
+bad_group = b"Bad tool group: "
+traceback = b"Traceback (most recent call last):\n"
+
+
+class Test_list_tools_no_tools_registered:
+    def test_help(self):
+        command = ["pbench-list-tools", "--help"]
+        out, err, exitcode = pytest.helpers.capture(command)
+        assert traceback not in err
+        assert empty == err
+        assert usage in out
+        assert exitcode == 0
+
+    def test_no_args(self, agent_config):
+        command = ["pbench-list-tools"]
+        out, err, exitcode = pytest.helpers.capture(command)
+        assert traceback not in err
+        assert empty == err
+        assert b"" == out
+        assert exitcode == 0
+
+
+class Test_list_tools_tools_registered:
+    @pytest.fixture
+    def tool(self, pbench_run):
+        p = pbench_run / "tools-v1-default" / "testhost.example.com"
+        p.mkdir(parents=True)
+        tool = p / "perf"
+        tool.touch()
```

Comment:
Are you sure that we want to be doing this "manually", instead invoking `pbench-register-tool`?  (Ditto for `tool_with_random_file()`.)

Reply:
Perhaps, although I also want to manufacture situations that `pbench-register-tool` and company cannot create. Ideally, the ToolGroup abstraction would be used (but it is read-only ATM) to manufacture the "legal" things and it's only the hinky ones that would have to be done by hand.